### PR TITLE
KBW-3685:  Phone verification carrier blacklist document

### PR DESCRIPTION
**Pull request by dkichler**
Support for storing blacklisted phone verification carriers in Couchbase for ease of updating

### CODE REVIEW

File: databases/business-sync/fragment-phone-verification-carrier-blacklist.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  channels: function(doc, oldDoc) {
+    // Only staff/service users can create, replace or delete payment attempts to prevent regular users from tampering
```
Comment (skwoka) [inline databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: :bureaucrat: copy-paste error: `payment attempts`

File: databases/business-sync/fragment-phone-verification-carrier-blacklist.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      view: staffChannel,
+      write: staffChannel
+    };
+  },
+  typeFilter: function(doc, oldDoc) {
+    return /^phoneVerificationCarrierBlacklist$/.test(doc._id);
+  },
+  propertyValidators: {
+    blacklist: {
+      // The ID of the business with which the payment attempt is associated
```
Comment (skwoka) [inline databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: The comment should be updated to reflect reality.

File: databases/business-sync/fragment-phone-verification-carrier-blacklist.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      type: 'array',
+      required: true,
+      minimumLength: 1,
```
Comment (skwoka) [inline databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: Should we make min length 0 to allow empty blacklist?

File: test/business-sync-phone-verification-carrier-blacklist.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+var businessSyncSpecHelperMaker = require('./helpers/business-sync-spec-helper-maker.js');
+var synctos = require('synctos');
+var testFixtureMaker = synctos.testFixtureMaker;
+var errorFormatter = synctos.validationErrorFormatter;
+
+describe('business-sync ohone verification carrier blacklist', function() {
```
Comment (skwoka) [inline test/business-sync-phone-verification-carrier-blacklist.js]: Typo: `ohone`

**Comment by skwoka** [comment]
Looks good. Left a few very minor comments.

File: databases/business-sync/fragment-phone-verification-carrier-blacklist.js (hunk 1 above)
Comment (dkichler) [reply databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: :doh:

File: databases/business-sync/fragment-phone-verification-carrier-blacklist.js (hunk 3 above)
Comment (dkichler) [reply databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: Hmm, I think I would rather have this guard against an accidental empty blacklist being posted - the document can always be deleted, or the feature flag turned off, if we wanted to disable the blacklist entirely.

Comment (skwoka) [reply databases/business-sync/fragment-phone-verification-carrier-blacklist.js]: 👍

**Pull request merged**